src/lib/reasoners/intervalCalculus.ml

```diff
@@ -2045,17 +2045,35 @@ let case_split env uf ~for_model =
       end
   | _ -> res
 
-let optimizing_split env uf opt_split =
+(* Helper function used in [optimizing_objective] to pick a value
+   for the polynomial [p] in its interval. We used this function in the case
+   the value produced by the optimization procedure doesn't satisfy some
+   constraints that involve strict inequalities or the problem is unbounded. *)
+let middle_value env ~to_max ty p bound =
+  let interval =
+    match MP0.find_opt p env.polynomes, bound with
+    | Some i, Some bound ->
+      begin
+        try
+          if to_max then
+            Intervals.new_borne_sup Ex.empty bound ~is_le:false i
+          else
+            Intervals.new_borne_inf Ex.empty bound ~is_le:false i
+        with Intervals.NotConsistent _ -> assert false
+      end
+    | Some i, None -> i
+    | None, _ -> Intervals.point Q.zero ty Ex.empty
+  in
+  let q = Option.get (Intervals.pick ~to_max interval) in
+  alien_of (P.create [] q ty)
+
+let optimizing_objective env uf Objective.Function.{ e; to_max; _ } =
   (* soundness: if there are expressions to optmize, this should be
      done without waiting for ~for_model flag to be true *)
-  let {Th_util.r = r; is_max = to_max; e; value; _ } = opt_split in
-  assert (match value with
-      | Unknown -> true
-      | _ -> false
-    );
+  let uf, _ = Uf.add uf e in
   let repr, _ = Uf.find uf e in
   let ty = E.type_info e in
-  let r1 = r in (* instead of repr, which may be a constant *)
+  let r1, _ = X.make e in (* instead of repr, which may be a constant *)
```

Comment:
Since `e` was added to the union-find this should be `Uf.make uf e` to avoid calling `X.make` twice.